_source/_posts/2021-09-12-spring-boot-spa.md

```diff
@@ -0,0 +1,686 @@
+> 1: Okta Spring Boot Starter
+> 2: Spring Boot
+> 3: JHipster
+> 4: Other
+Enter your choice [Other]: 4
+Redirect URI
+Common defaults:
+ Spring Security - http://localhost:8080/login/oauth2/code/okta
+ JHipster - http://localhost:8080/login/oauth2/code/oidc
+Enter your Redirect URI [http://localhost:8080/callback]: 
+Enter your Post Logout Redirect URI [http://localhost:8080/]: 
+Configuring a new OIDC Application, almost done:
+Created OIDC application, client-id: 02b4cj4ys0G92cXYI4x7
+
+Okta application configuration has been written to: /home/andrewcarterhughes/Development/okta/2021/spring-boot-spa/spring-boot-spa/.okta.env
+```
+
+This creates a configuration file, `.okta.env` that you can source to configure the application.
+
+```bash
+source .okta.env
+```
+
+**You will need to source this file each time you open a new console** to run the application. 
+
+In a production scenario, you will need to ensure that these environment variables are loaded into the environment in which the Spring Boot application is running. **For security, it's also important that the file, and the client secret in particular, is not checked into source control or otherwise leaked.** Thus you may want to add `.okta.env` to the `.gitignore` of any similar projects you create (as I have here).
+
+The `.okta.env` file is simply a script that exports the necessary properties.
+
+```bash
+export OKTA_OAUTH2_ISSUER="https://dev-123456.okta.com/oauth2/default"
+export OKTA_OAUTH2_CLIENT_SECRET="M-alsjdfoiw932owur08u2093u0932u092jlj"
+export OKTA_OAUTH2_CLIENT_ID="02b4cj4ys0G92cXYI4x7"
+```
+
+You now have a working, secure resource server. Go ahead and run the application.
+
+```bash
+./mvnw spring-boot:run
+```
+
+You should get a lot of console output that ends with the following.
+
+```bash
+2021-09-01 11:49:42.136  INFO 370077 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8082 (http) with context path ''
+2021-09-01 11:49:42.143  INFO 370077 --- [           main] com.okta.springbootspa.DemoApplication   : Started DemoApplication in 2.021 seconds (JVM running for 2.199)
+```
+
+## Use The Vue CLI To Bootstrap The Client App
+
+The next step is to build the front-end, client application. The application will use Vue and the Okta Sign-In Widget to implement a secure client application. The Okta Sign-In Widget allows you to easily add Single Sign-On (SSO) and a secure login form to any application.
+
+The instructions below for integrating the Okta Sign-In widget with Vue are largely pulled from [this article](https://developer.okta.com/code/vue/okta_vue_sign-in_widget/#add-an-openid-connect-client-in-okta) on the Okta website, which is a great reference if you have any questions.
+
+Create the app using the Vue CLI. Run the following command from the project root directory (the directory above the spring boot project directory). It will create a new directory, `okta-app`, that will contain the client application.
```

Comment:
```suggestion
Create the app using the Vue CLI. Run the following command from the project root directory (the directory above the spring boot project directory). It will create a new directory, `okta-app`, which will contain the client application.
```
after a comma 'which' works better